FutureFeatures.md

```diff
@@ -273,14 +273,19 @@ quadruple precision.
 
 ## Floating point library intrinsics
 
-These operations aren't needed because they can be implemented in WebAssembly
-code and linked into WebAssembly modules as at small size cost, and this avoids
-a non-trivial specification burden of their semantics/precision. Adding these
-intrinsics would allow for better high-level backend optimization of these
-intrinsics that require builtin knowledge of their semantics. On the other
-hand, a code generator may continue to statically link in its own
-implementation since this provides greater control over precision/performance
-tradeoffs.
+These operations aren't needed for the MVP because they can be implemented in
+WebAssembly code and linked into WebAssembly modules at small size cost (as is
+traditionally done through C libraries such as libm). This avoids a non-trivial
+specification burden for their semantics, precision, and performance.
+
+[Dynamic linking](FutureFeatures.md#dynamic-linking) will allow caching of libm,
+making this sharing cost trivial.
+
+Adding these intrinsics would potentially allow for better high-level backend
+optimization of these intrinsics that require builtin knowledge of their
+semantics. WebAssembly may support these operations in the future if data shows
+it would be useful. The rounding behavior of these operations would need
+clarification.
```

Comment:
If we're going to discuss this topic in detail, we should also mention an alternative way to achieve the same benefits, and much more: develop an optimization/JIT library, which could run as a library inside WebAssembly, and would be useful for general-purpose JITs of all kinds. Such a library could much more freely offer a much greater degree of nondeterminism because it wouldn't be baked into the platform, so it could theoretically offer services like:
 - math functions that aren't always perfectly rounded
 - lots of fast-math optimizations
 - evaluation of constant expressions at greater intermediate precision
 - optimizations that depend on knowing that integer arithmetic doesn't overflow
 - optimizations that depend on knowing that all loops either have side effects or terminate
 - higher-level semantics
 - many others!

One possibility here would be to compile and use LLVM inside WebAssembly, and I expect some people will do exactly this. There's also room for simpler and lighter-weight solutions as well; code for a simple IR, with basic inlining and constant folding (including of trigonometic and exponential functions!) would be quite simple.

And the great thing about a library approach is that one can depend on getting the same optimizations in all implementations. There's no need to make sure that every browser does the same inlining and thus achieves the same math function constant folding. One's optimization library either does it or it doesn't, everywhere.